x-pack/legacy/plugins/apm/public/components/app/TraceLink/index.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import url from 'url';
+import styled from 'styled-components';
+import { Redirect } from 'react-router-dom';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import { QueryParams } from '../../shared/Links/apm/ExternalLinks';
+import { TRACE_ID } from '../../../../common/elasticsearch_fieldnames';
+
+const CentralizedContainer = styled.div`
+  height: 100%;
+  display: flex;
+`;
+
+type DateRange = Partial<QueryParams>;
+
+const getDateRange = ({ rangeFrom, rangeTo }: DateRange) =>
+  rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {};
+
+const redirectToTransactionDetailPage = ({
+  transaction,
+  rangeFrom,
+  rangeTo
+}: { transaction: Transaction } & DateRange) =>
```

Comment:
```suggestion
}: { transaction: Transaction, rangeFrom: string, rangeTo: string }) =>
```

Reply:
Ok, so I change it to:

```js
const redirectToTransactionDetailPage = ({
  transaction,
  rangeFrom,
  rangeTo
}: {
  transaction: Transaction;
  rangeFrom?: string;
  rangeTo?: string;
}) =>
```
Even though there is no chance of rangeFrom and rangeTo to be undefined, they are still defined as optional here:
```js
export type IUrlParams = {
  detailTab?: string;
  end?: string;
  errorGroupId?: string;
  flyoutDetailTab?: string;
  kuery?: string;
  environment?: string;
  rangeFrom?: string;
  rangeTo?: string;
```

In the worst scenario, the user will be redirected to `services/opbeans-ruby/transactions/view?rangeFrom=&rangeTo=&traceId=c130f46cb7e91675cf0be1a8fdd98dcc&transactionId=15e9fcc39fb5b4ce&transactionName=Api::OrdersController%23index&transactionType=request`